web/src/components/ZoneName.tsx

```diff
@@ -5,7 +5,7 @@ import { CountryFlag } from './Flag';
 
 export function ZoneName({ zone, textStyle }: { zone: string; textStyle?: string }) {
   return (
-    <div className="flex items-center">
+    <div className="flex min-w-0 items-center">
```

Comment:
This is so the truncation works properly. Basically the min width needs to be anything other than auto.